### ISSUE TITLE
bstd.UnmarshalByteSlice panics

### ISSUE THREAD

**Issue by fujistick** [deneonet/benc#1]
**Describe the bug**
`bstd.UnmarshalByteSlice` panics where len(slice) < 4 or returns incorrect slice with larger slice sizes.

**To Reproduce**
Run the following code.  Note how it panics with `panic: runtime error: slice bounds out of range [4:3]`
```
inByteSlice := []byte{1, 2, 3}
size := bstd.SizeByteSlice(inByteSlice)

buf := make([]byte, size)
bstd.MarshalByteSlice(0, buf, inByteSlice)

_, outByteSlice, err := bstd.UnmarshalByteSlice(0, buf) // <-- this panics
```

**Expected behavior**
Not to panic, and to return the correct outByteSlice.

**Golang version:**
go version go1.21.6 darwin/arm64

**BENC version:**
v1.0.5

**Additional context**
The fix looks to be replacing:
```
func UnmarshalByteSlice(n int, b []byte) (int, []byte, error) {
	// ...
	bs := b[n+4 : n+size]
	// ...
}
```
with:
```
func UnmarshalByteSlice(n int, b []byte) (int, []byte, error) {
	// ...
	bs := b[n+4 : n+4+size]
	// ...
}
```

**Comment by deneonet** [comment]
A silly mistake I made, thanks for finding it. I'll fix it as soon as possible.

**Comment by deneonet** [comment]
Fixed in v1.0.6
